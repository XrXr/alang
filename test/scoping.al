main :: proc () {
    a := "outter a\n"
    c := "sea"
    puts(a)
    if true {
        a := "inner a\n"
        puts(a)
        a = "modified inner a\n"
        b := "shouldn't print"

        if true {
            puts(a)
            if true {
                b = "must go deeper!\n"
                if true {
                    puts(b)
                    c = "reaching out\n"
                }
            }
        }
    }
    puts(a)
    puts(c)
}
