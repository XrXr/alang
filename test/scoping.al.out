outter a
inner a
modified inner a
must go deeper!
outter a
reaching out
